docs/modules/ROOT/pages/explanations/decisions/comp-function-error-handling.adoc

```diff
@@ -0,0 +1,71 @@
+= Composition Function Error Handling
+
+In March 2024 we had a severe outage of VSHN PostgreSQL instances.
+This was due to several unfortunate events including our poor handling of errors in composition functions.
+This document will discuss in detail the problem we face with composition functions when errors are not properly handled.
+
+== Problems
+
+* We don't manage errors in our composition functions (CFs) and we assume Crossplane will take care of everything.
```

Comment:
I mean... we do manage errors, but we're not doing a very good job :D

Reply:
you are right, added `properly`